app/controllers/active_storage/blobs/proxy_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# Proxy files through application with authentication.
+# This avoids making the S3 bucket where files are stored public
+
+# Overwriting
+# https://github.com/rails/rails/blob/6-1-stable/activestorage/app/controllers/active_storage/blobs/proxy_controller.rb
+class ActiveStorage::Blobs::ProxyController < ActiveStorage::BaseController
+  include ActiveStorage::SetBlob
+  include ActiveStorage::SetHeaders
+
+  def show
+    redirect_to('/') && return unless current_user&.id == @blob&.attachments&.first&.record&.user_id
```

Comment:
Assuming that if I'm a user that's not logged in, then `current_user` would be `nil`. If any of the safe navigation for `@blob` fails and returns nil AND I'm not logged in, then `nil == nil` would be true. 

Are there any cases where we should expect `@blob&.attachments&.first&.record&.user_id` to return `nil`? If so, we should probably at least check if `current_user` is not nil.

Reply:
I've added protection against this.